react/utils/components/index.ts

```diff
@@ -283,12 +285,24 @@ const keepTheBlanks = (
   )
 }
 
+export const appendInvisibleCharacter = (text: string) =>
+  text + INVISIBLE_CHARACTER
+
+const removeInvisibleCharacter = (text: string) => {
+  const lastIndex = text.length - 1
+
+  if (text[lastIndex] !== INVISIBLE_CHARACTER) {
+    return text
+  }
+
+  return text.slice(0, lastIndex)
```

Comment:
Are you appending more than one `INVISIBLE_CHARACTER` to the text? If you're appending `INVISIBLE_CHARACTER` on every onChange, you could end up with the string:
`lalal${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}${INVISIBLE_CHARACTER}`, couldn't you?

If so, the `text.length - 1` isn't reliable, I'd use `indexOf` to get the first (in line 292). Also, I would change `.slice` to `.replace` here, as it's safer (to cases like multiple appends, and other cases like `a${INVISIBLE_CHARACTER}b${INVISIBLE_CHARACTER}`). What do you think?

Reply:
There's always at most only one invisible character.

The `newValue` from [this line](https://github.com/vtex-apps/admin-pages/pull/279/files#diff-aebfa0473e42ecce8b1a6dec918ef42fR47) refers to the new value in the input, and it never contains an invisible character – unless the user added it himself, in which case we shouldn't interfere.

Reply:
Come to think of it, [`removeInvisibleCharacter`](https://github.com/vtex-apps/admin-pages/pull/279/files#diff-9b3ddc4469aa6462dfe328035ea0c290R291) will always remove the last invisible character, even if it was added by the user. I don't think people will actually put trailing invisible characters in their strings, but you never know; I like to believe we'll have more efficient ways to update the messages before this becomes an issue.

What do you think?

Reply:
> I don't think people will actually put trailing invisible characters in their strings, but you never know

I agree and had the feeling that you'd comment this ⛏ . What I think could happen is the user copy+pasting from the iframe and appending multiple invisible characters, ergo, my suggestion to use `.replace`:

<img width="1291" alt="Screen Shot 2019-09-17 at 12 03 37" src="https://user-images.githubusercontent.com/10400340/65053992-3c58ce80-d943-11e9-8423-118882473e77.png">

Reply:
Hmm, the user would have to copy from the iframe while editing a block, but yes, that could happen. Okay, I'll use `replace` instead!